### DISCUSSION TITLE
How to refer to a .sol file inside the test file with correct syntax?

### DISCUSSION THREAD

**Discussion by sushrut-d07** [Cyfrin/foundry-full-course-cu#2716] [Q&A] [answered]
Only able to use Raffle as an object not as a file (screenshot)
![object](https://github.com/user-attachments/assets/48ea731a-8ab9-4bfc-a87f-dc83248e52d7)

This is the error 
`Error: 
Compiler run failed:
Error (9582): Member "Raffle_NotEnoughEntryFeesSent" not found or not visible after argument-dependent lookup in type(contract Raffle).
  --> test/unit/RaffleTest.t.sol:44:25:
   |
44 |         vm.expectRevert(Raffle.Raffle_NotEnoughEntryFeesSent.selector);
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^`
   
   PS : These are my imports
   `import {DeployRaffle} from "../../script/DeployRaffle.s.sol";
import {Raffle} from "../../src/Raffle.sol";
import {Test, console} from "forge-std/Test.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";`

**Comment by EngrPips** [comment]
Hello @sushrut-d07, Please show the `Raffle_NotEnoughEntryFeesSent` in your `Lottery` contract.

**Reply by sushrut-d07** [reply]
its outside my contract, after imports like this.
![error](https://github.com/user-attachments/assets/b1106954-3bc0-4ae9-8733-4a89968511ec)

**Reply by EngrPips** [reply, marked as answer]
Put them inside your contract, then retry. Didn't try with it being outside.

**Reply by sushrut-d07** [reply]
dont we follow this order to write our code
Pragma statements
Import statements
Events
Errors
Interfaces
Libraries
Contracts

**Reply by EngrPips** [reply]
There general standards but then you can write your contract in any other you want as far it works correctly.
